packages/leap/lib/src/leap_map.dart

```diff
@@ -41,69 +44,92 @@ class LeapMap extends PositionComponent with HasGameRef<LeapGame> {
     return super.onMount();
   }
 
-  /// Convenience method for accessing Tiled layers in the [tiledMap]
-  T? getTileLayer<T extends Layer>(String name) {
-    return tiledMap.tileMap.getLayer<T>(name);
+  /// Convenience method for accessing Tiled layers in the [tiledMap].
+  T getTileLayer<T extends Layer>(String name) {
+    final layer = tiledMap.tileMap.getLayer<T>(name);
+    if (layer != null) {
+      return layer;
+    }
+
+    Error.throwWithStackTrace(
+      Exception(
+        'Layer $name not found, check the Tiled map for the correct name.',
+      ),
+      StackTrace.current,
+    );
```

Comment:
How is this different from `throw Exception('Layer $name not found, check the Tiled map for the correct name.')`?

If you're just using `StackTrace.current` I don't think there is any reason to use `Error.throwWithStackTrace`, right?